src/connection/mod.rs

```diff
@@ -0,0 +1,160 @@
+/*
+
+* This file is part of Astarte.
+*
+* Copyright 2023 SECO Mind Srl
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use tokio::sync::RwLock;
+
+use crate::{
+    interface::{
+        mapping::path::MappingPath,
+        reference::{MappingRef, ObjectRef},
+        Ownership,
+    },
+    interfaces::Interfaces,
+    shared::SharedDevice,
+    store::{wrapper::StoreWrapper, PropertyStore, StoredProp},
+    types::AstarteType,
+    validate::{ValidatedIndividual, ValidatedObject},
+    Interface, Timestamp,
+};
+
+pub mod mqtt;
+
+pub(crate) struct Introspection {
```

Comment:
Why should introspection contain the `device_properties`? Introspection is a set of supported interfaces declared by the device.

Reply:
This struct got moved to the `mqtt` module, strictly speaking, yes the introspection is just a set of supported interfaces but, in the case of the MQTT transport, along with the introspection we also subscribe to server interfaces and send the stored device properties... This struct wraps up all the data that we use while connecting.

Reply:
I have renamed the struct, it is now called `SessionData`.